Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -936,42 +938,42 @@ google_firestore_v1_Target_QueryTarget Serializer::EncodeQueryTarget(
   }
 
   // Encode the filters.
-  const auto& filters = query.filters();
+  const auto& filters = target.filters();
   if (!filters.empty()) {
     result.structured_query.where = EncodeFilters(filters);
   }
 
-  const auto& orders = query.order_bys();
+  const auto& orders = target.order_bys();
   if (!orders.empty()) {
     result.structured_query.order_by_count = CheckedSize(orders.size());
     result.structured_query.order_by = EncodeOrderBys(orders);
   }
 
-  if (query.limit() != Query::kNoLimit) {
+  if (target.limit() != Target::kNoLimit) {
     result.structured_query.has_limit = true;
-    result.structured_query.limit.value = query.limit();
+    result.structured_query.limit.value = target.limit();
   }
 
-  if (query.start_at()) {
-    result.structured_query.start_at = EncodeBound(*query.start_at());
+  if (target.start_at()) {
+    result.structured_query.start_at = EncodeBound(*target.start_at());
   }
 
-  if (query.end_at()) {
-    result.structured_query.end_at = EncodeBound(*query.end_at());
+  if (target.end_at()) {
+    result.structured_query.end_at = EncodeBound(*target.end_at());
   }
 
   return result;
 }
 
-Query Serializer::DecodeQueryTarget(
+Target Serializer::DecodeQueryTarget(
     nanopb::Reader* reader,
     const google_firestore_v1_Target_QueryTarget& proto) const {
   // The QueryTarget oneof only has a single valid value.
   if (proto.which_query_type !=
       google_firestore_v1_Target_QueryTarget_structured_query_tag) {
     reader->Fail(
         StringFormat("Unknown query_type: %s", proto.which_query_type));
-    return InvalidQuery();
+    return InvalidTarget();
```

Comment:
This can just be `return {}`, which we've used elsewhere for returning a default value after an error.

Here and below.